lib/jekyll/filters/url_filters.rb

```diff
@@ -23,9 +23,9 @@ def absolute_url(input)
       # Returns a URL relative to the domain root as a String.
       def relative_url(input)
         return if input.nil?
-        return ensure_leading_slash(input.to_s) if sanitized_baseurl.nil?
+        parts = [sanitized_baseurl, input]
         Addressable::URI.parse(
-          ensure_leading_slash(sanitized_baseurl) + ensure_leading_slash(input.to_s)
+          parts.compact.map { |part| ensure_leading_slash(part.to_s) }.join
```

Comment:
Because `parts.compact` returns a copy of `parts`: `parts'`
and `map` will return another copy of `parts'`: `parts''`

in effect loading memory with unnecessary objects (at each iteration in the example cited in this PR)